### PR TITLE
librados,osdc: kill ack vs commit distinction

### PR DESCRIPTION
**Pull request by liewegas**
Remove duplicate callbacks.  Simplify code.

### CODE REVIEW

**Comment by liewegas** [comment]
@jdurgin one additional patch to Objecter

**Comment by jdurgin** [comment]
removing oncommit_sync looks good

File: src/client/Client.cc [hunk 1]
```diff
@@ -12088,13 +12069,11 @@ int Client::ll_write_block(Inode *in, uint64_t blockid,
   if (true || sync) {
     /* if write is stable, the epilogue is waiting on
      * flock */
-    onack = new C_NoopContext;
```
Comment (jcsp) [inline src/client/Client.cc]: This was the last user of C_NoopContext so could remove it from Context.h too

**Comment by jcsp** [comment]
Looks fine to me!

**Pull request merged**